### PR TITLE
Change

### PR DESCRIPTION
**Pull request by akshat01112001**

### CODE REVIEW

File: .husky/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+_
```
Comment (jackfranklin) [inline .husky/.gitignore]: Do you know what caused this change?

File: package.json [hunk 2]
```diff
@@ -109,6 +110,7 @@
     "standard-version": "^9.0.0",
     "text-diff": "^1.0.1",
     "ts-node": "^9.0.0",
+    "typedoc": "^0.20.32",
```
Comment (jackfranklin) [inline package.json]: typedoc is different to TSDoc (confusingly!) so I don't think we want this. Why did you add it?

File: src/common/Browser.ts [hunk 3]
```diff
@@ -412,7 +412,7 @@ export class Browser extends EventEmitter {
       url: 'about:blank',
       browserContextId: contextId || undefined,
     });
-    const target = await this._targets.get(targetId);
+    const target = this._targets.get(targetId);
```
Comment (jackfranklin) [inline src/common/Browser.ts:418]: Nice spot!

File: src/common/Connection.ts [hunk 4]
```diff
@@ -67,8 +67,8 @@ export class Connection extends EventEmitter {
   }
 
   /**
-   * @param {string} sessionId
-   * @returns {?CDPSession}
+   * @param sessionId - Gets Session ID
+   * @returns - Gets CDPSession
```
Comment (jackfranklin) [inline src/common/Connection.ts]: You don't need a dash after `@returns` - see https://tsdoc.org/pages/tags/returns/

So this should be

```
@returns the current CDP session if it exists

File: src/common/Connection.ts [hunk 5]
```diff
@@ -167,8 +167,8 @@ export class Connection extends EventEmitter {
   }
 
   /**
-   * @param {Protocol.Target.TargetInfo} targetInfo
-   * @returns {!Promise<!CDPSession>}
+   * @param targetInfo - Gets targetInfo
+   * @returns sessionID
```
Comment (jackfranklin) [inline src/common/Connection.ts:179]: try and write these like a sentence, so this might be:

```
@returns the CDP session that is created
```

File: src/common/Connection.ts [hunk 6]
```diff
@@ -167,8 +167,8 @@ export class Connection extends EventEmitter {
   }
 
   /**
-   * @param {Protocol.Target.TargetInfo} targetInfo
-   * @returns {!Promise<!CDPSession>}
+   * @param targetInfo - Gets targetInfo
```
Comment (jackfranklin) [inline src/common/Connection.ts]: targetInfo here is a param the user provides, so "Gets targetInfo" doesn't quite fit. If you're not sure what to document for this param, just leave it as:

```
@param targetInfo
```

File: src/common/EventEmitter.ts [hunk 7]
```diff
@@ -33,7 +33,7 @@ export interface CommonEventEmitter {
  * {@link EventEmitter.off | off} to bind
  * and unbind to event listeners.
  *
- * @public
+ * @internal
```
Comment (jackfranklin) [inline src/common/EventEmitter.ts]: why did you change this? We do want to document this class so I don't believe this should be internal.

File: src/common/ExecutionContext.ts [hunk 8]
```diff
@@ -33,7 +33,7 @@ const SOURCE_URL_REGEX = /^[\040\t]*\/\/[@#] sourceURL=\s*(\S*?)\s*$/m;
  *   {@link https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe |
  *   frame } has "default" execution context that is always created after frame is
  *   attached to DOM. This context is returned by the
- *   {@link frame.executionContext()} method.
+ *   {@link frame.executionContext()| method} method.
```
Comment (jackfranklin) [inline src/common/ExecutionContext.ts]: this will case the rendered text to read `method method` - the bit after the `|` is the text used. Why did you make this change?

File: src/common/FileChooser.ts [hunk 9]
```diff
@@ -33,7 +33,8 @@ import { assert } from './assert.js';
  * ```
  * **NOTE** In browsers, only one file chooser can be opened at a time.
  * All file choosers must be accepted or canceled. Not doing so will prevent
- * subsequent file choosers from appearing.
+ * subsequent file choosers from app/earing.
+ * @internal
```
Comment (jackfranklin) [inline src/common/FileChooser.ts]: I think this should actually be `@public` as we want it documented to the user.

File: src/common/Page.ts [hunk 10]
```diff
@@ -283,7 +283,7 @@ export const enum PageEmittedEvents {
    * Emitted when a page issues a request and contains a {@link HTTPRequest}.
    *
    * @remarks
-   * The object is readonly. See {@Page.setRequestInterception} for intercepting
+   * The object is readonly. See /{@Page setRequestInterception \} for intercepting
```
Comment (jackfranklin) [inline src/common/Page.ts]: why did you add the slashes here?

**Review by jackfranklin — COMMENTED** [review]
Thanks for your PR! I've left some comments with some questions, if you wouldn't mind taking another look 👍

**Comment by akshat01112001** [comment]
I have updated the changes. Have a look.

Cheers,
Akshat

File: src/common/Page.ts [hunk 11]
```diff
@@ -283,7 +283,7 @@ export const enum PageEmittedEvents {
    * Emitted when a page issues a request and contains a {@link HTTPRequest}.
    *
    * @remarks
-   * The object is readonly. See {@Page.setRequestInterception} for intercepting
+   * The object is readonly. See {@Page setRequestInterception} for intercepting
```
Comment (jackfranklin) [inline src/common/Page.ts]: We still need the dot here - this should be `@Page.setRequestInterception` so TSDoc links to the right method

File: src/common/ExecutionContext.ts (hunk 8 above)
Comment (jackfranklin) [reply src/common/ExecutionContext.ts]: @akshat01112001 please could you undo this change?

Comment (akshat01112001) [reply src/common/ExecutionContext.ts]: I have undone this change

**Review by jackfranklin — APPROVED** [review]
